models/DataObject/ClassDefinition/Data/Block.php

```diff
@@ -207,6 +207,12 @@ public function getDataFromResource($data, $object = null, $params = [])
         if ($data) {
             $count = 0;
 
+            //Fix old serialized data protected properties with \0*\0 prefix
+            //https://github.com/pimcore/pimcore/issues/9973
+            $data = preg_replace_callback ( '!s:(\d+):" \* (.*?)";!', function($match) {
```

Comment:
@dvesh3 should we maybe check with `str_contains()` first, as it is much faster than running the regex on any  value?